src/components/ChatList/GroupChatChannelItem.tsx

```diff
@@ -20,23 +21,33 @@ const GroupChatChannelItem = (props: ChannelItemProps) => {
   const unreadCount = groupChat?.unreadCount;
   const hasBadge = unreadCount > 0;
 
-  const description = groupChat?.rawJson?.firstMessage ?? groupChat?.rawJson?.message;
-  const isSystemDescription =
-    description?.isSystem ||
-    description?.type === 'system' ||
-    description?.message?.type === 'system';
-  const isDeletedMessage = description?.type === 'deleted';
-  const sender = description?.user?.username ?? description?.user?.name;
-  const isMeAsSender =
-    description?.user?.id === signedProfileId || description?.user?.isMe || !description;
+  const isMe = groupChat?.lastUpdatedBy === signedProfileId;
+  const firstMessage = groupChat?.firstMessage;
+  const firstMessageText = firstMessage?.message;
+  let description = groupChat?.description;
 
-  let channelDescription = groupChat?.description;
-  if (!isSystemDescription && description !== channelDescription) {
-    channelDescription = isMeAsSender
-      ? `You: ${channelDescription}`
-      : `${sender}: ${channelDescription}`;
+  if (['deleted', 'notification-deleted'].includes(groupChat?.firstMessage?.type || '')) {
+    description = DELETED_MESSAGE_TEXT;
+  } else if (groupChat?.firstMessage?.type === 'system') {
+    description = firstMessageText || '';
+  } else if (isMe) {
+    description = `You: ${description}`;
+  } else if (!isMe) {
+    description = `${groupChat?.user?.username}: ${description}`;
   }
-  if (isDeletedMessage) channelDescription = '';
+
+  const getMessageText = (): React.ReactNode => {
+    if (groupChat?.firstMessage?.attachmentJson?.length) {
+      return (
+        <Text>
+          <Text>{isMe ? 'You: ' : `${groupChat?.user?.username}: `}</Text>
+          <Text style={{fontStyle: 'italic'}}>Sent media 🏞️</Text>
+        </Text>
+      );
+    }
+
+    return `${isMe ? 'You: ' : `${groupChat?.user?.username}: `}${description}`;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for constructing the description is a bit complex and could be simplified. Also, it seems like there's some redundancy in checking `isMe` multiple times. Here's a suggestion to simplify this part:

```diff
-  const isMe = groupChat?.lastUpdatedBy === signedProfileId;
-  const firstMessage = groupChat?.firstMessage;
-  const firstMessageText = firstMessage?.message;
-  let description = groupChat?.description;
-
-  if (['deleted', 'notification-deleted'].includes(groupChat?.firstMessage?.type || '')) {
-    description = DELETED_MESSAGE_TEXT;
-  } else if (groupChat?.firstMessage?.type === 'system') {
-    description = firstMessageText || '';
-  } else if (isMe) {
-    description = `You: ${description}`;
-  } else if (!isMe) {
-    description = `${groupChat?.user?.username}: ${description}`;
-   }
-
-  const getMessageText = (): React.ReactNode => {
-    if (groupChat?.firstMessage?.attachmentJson?.length) {
-      return (
-        <Text>
-          <Text>{isMe ? 'You: ' : `${groupChat?.user?.username}: `}</Text>
-          <Text style={{fontStyle: 'italic'}}>Sent media 🏞️</Text>
-        </Text>
-      );
-    }
-
-    return `${isMe ? 'You: ' : `${groupChat?.user?.username}: `}${description}`;
-  };
+  const isMe = groupChat?.lastUpdatedBy === signedProfileId;
+  const firstMessage = groupChat?.firstMessage;
+  const firstMessageText = firstMessage?.message;
+  const prefix = isMe ? 'You: ' : `${groupChat?.user?.username}: `;
+  let description = groupChat?.description;
+
+  switch (firstMessage?.type) {
+    case 'deleted':
+    case 'notification-deleted':
+      description = DELETED_MESSAGE_TEXT;
+      break;
+    case 'system':
+      description = firstMessageText || '';
+      break;
+    default:
+      description = `${prefix}${description}`;
+  }
+
+  const getMessageText = (): React.ReactNode => {
+    if (groupChat?.firstMessage?.attachmentJson?.length) {
+      return (
+        <Text>
+          <Text>{prefix}</Text>
+          <Text style={{fontStyle: 'italic'}}>Sent media 🏞️</Text>
+        </Text>
+      );
+    }
+
+    return description;
+  };
```
This way, we only check `isMe` once and construct the prefix accordingly. Then we use a switch statement to handle different types of messages, which makes it easier to add more types in the future.



<!-- This is an auto-generated comment by OSS CodeRabbit -->